### PR TITLE
Add sql database

### LINKED ISSUE #5: Quota storage not thread safe
May not be that important since this is a demo, but you could lose data if you get parallel requests:
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/database.py#L23-L26

Maybe just use something like a SQLite db which is also really easy to use in Python

### LINKED ISSUE #14: Route `GET /quota` delivers all data
The route should not deliver all quotas from the json file. Instead it should only deliver the specified quotas.

https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L80

### LINKED ISSUE #15: Route `PUT /quota` handles data incorrectly
Append doesn't seem right. The quota should be replaced if a quota with scope and feature already exists:
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L93

### LINKED ISSUE #17: Quota for different feature overwrite each other
Doesn't this code need to take `features` into account as well?
I can set different quotas for the same scope but different features, can't I?

https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L121-L125

Also… a database would make something like this quite a bit easier :D

### LINKED ISSUE #19: Data doesn't meet the specification
Some mock data doesn't meet the specification. The following places are related.

Some quotas don't have a used value as they are common quotas like course quota:
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L91

We didn't specify a default type. It shouldn't be set by the toolbox:
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L92

Loads individual course user quotas which should not:
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L102-L103

Only scopes "course" and "course-user" are supported (validation check):
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L122

Also loads the common course user quota for this course:
https://github.com/Odrec/mockup_api/blob/b12a9712e1499802cbac8581b802cb1f129c5d97/main.py#L136

### PR DESCRIPTION
**Pull request by dennis531**
Close #5
Close #14
Close #15
Close #17
Close #19

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -45,109 +68,71 @@ async def get_metadata():
...
-    course_quotas = [validate_quota(q) for q in quotas if
-                     q['scope'] in ['course', 'course-user'] and q.get('course_id') == course_id]
-    if not course_quotas:
-        raise HTTPException(status_code=404, detail="Course quota not found")
-    if with_user_quotas:
-        user_quotas = [validate_quota(q) for q in quotas if
-                       q['scope'] == 'course-user' and q.get('course_id') == course_id]
-        course_quotas.extend(user_quotas)
-    return course_quotas
+async def get_course_quota(course_id: str, db: Session = Depends(get_db)):
```
Comment (Odrec) [inline main.py:109]: did we get rid of the with_user_quotas parameter? Do we still need it?

**Review by Odrec — COMMENTED** [review]
I looked at it quickly and I think everything is fine. The only doubt I have was what happened to the with_user_quotas parameter in get_course_quota.

File: main.py (hunk 1 above)
Comment (dennis531) [reply main.py:109]: The user quota parameter was replaced by the route `/quota/course/{course_id}/user` which provides the same function in the specification.

Comment (Odrec) [reply main.py:109]: Ah, understood. Then I think everything is fine. I'll approve the merge.

**Pull request merged**